src/beaconer.rs

```diff
@@ -164,6 +168,83 @@ impl Beaconer {
             }
         };
 
+        // check if hash of witness is below the "difficulty threshold" for a secondary beacon
+        // TODO provide a way to get this difficulty threshold from eg. the entropy server
+        let buf = Sha256::digest(report.encode_to_vec()).to_vec();
+        let threshold = BigInt::parse_bytes(
+            b"11388830933659919894162346859235831137017100287433801699915283717462644789984",
+            10,
+        )
+        .unwrap();
+        // compare the hash of the witness report as a bigint to the difficulty threshold
+        // this is sort of a bitcoin-esque proof of work check insofar as as we're looking
+        // for hashes under a certain value. Because of the time constraints involved this
+        // should not be a 'mineable' check, but it provides a useful probabalistic way to
+        // allow for verifiable secondary beacons without any coordination.
+        let factor = BigInt::from_bytes_be(Sign::Plus, &buf);
+        if let (Some(region_params), true) = (&self.region_params, factor < threshold) {
```

Comment:
Pull the region param and early return out sooner? or perhaps turn this into a function like 

`async fn derive_secondary_beacon(&self..) -> Result<Option<Beacon>>`

Which would give you a single error to match on, log and return instead of the many matches in this caller.. Then Option<Beacon> gets you a beacon or not if the threshold wasn't met